openedx/core/djangoapps/user_authn/views/password_reset.py

```diff
@@ -659,3 +662,54 @@ def password_reset_token_validate(request):
         AUDIT_LOG.exception("Invalid password reset confirm token")
 
     return JsonResponse({'is_valid': is_valid})
+
+
+def _is_token_available(**kwargs):
+    """
+    Helper function to test that token
+    is available in view reset url
+    """
+    uidb36 = kwargs.get('uidb36')
+    token = kwargs.get('token')
+
+    if not uidb36 or not token:
+        return False
+    try:
+        _ = base36_to_int(uidb36)
+    except ValueError:
+        return False
+    return True
+
+
+@require_POST
+@ensure_csrf_cookie
+def password_reset_logistration(request, **kwargs):
+    """Reset learner password using passed token and new credentials"""
+
+    reset_status = False
+    uidb36 = kwargs.get('uidb36')
+    token = kwargs.get('token')
+
+    if not _is_token_available(**kwargs):
+        AUDIT_LOG.exception("Invalid password reset confirm token")
+        return JsonResponse({'reset_status': reset_status})
+
+    uid_int = base36_to_int(uidb36)
+    request.POST = request.POST.copy()
+    request.POST['new_password1'] = normalize_password(request.POST['new_password1'])
+    request.POST['new_password2'] = normalize_password(request.POST['new_password2'])
+
+    password = request.POST['new_password1']
+    try:
+        user = User.objects.get(id=uid_int)
+        validate_password(password, user=user)
+        form = SetPasswordForm(user, request.POST)
+        if default_token_generator.check_token(user, token) and form.is_valid():
```

Comment:
Can we do this check a bit early so that If the token isn't valid we don't have to do all the above processing?